File: Data Structures & Algorithms/spiral-matrix/submission-0.cpp
class Solution {
public:
    vector<int> spiralOrder(vector<vector<int>>& matrix) {
        int T = 0, B = matrix.size() - 1, L = 0, R = matrix[0].size() - 1;
        vector<int> res;

        while (L <= R && T<= B){
            //L -> R
            for (int i = L; i <= R; i++){
                res.push_back(matrix[T][i]);
            }
            T++;
            //T -> B
            for (int i = T; i <= B; i++){
                res.push_back(matrix[i][R]);
            }
            R--;
            //R -> L
            if (T > B) break;
            for (int i = R; i >= L; i--){
                res.push_back(matrix[B][i]);
            }
            B--;
            //B->T
            if (L > R) break;
            for (int i = B; i >= T; i--){
                res.push_back(matrix[i][L]);
            }
            L++;
        }
        return res;
        
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = 0

        def dfs(node, cur_max):
            nonlocal res
            if not node:
                return
            
            if node.val >= cur_max:
                res += 1
                cur_max = node.val
            
            dfs(node.left, cur_max)
            dfs(node.right, cur_max)
        dfs(root, -float('inf'))
        return res

            
            
        